src/libsystemd-network/sd-dhcp-client.c

```diff
@@ -1509,16 +1526,18 @@ static int client_parse_message(
         }
 
         r = dhcp_option_parse(message, len, dhcp_lease_parse_options, lease, &error_message);
+        if (client->bootp && r == -ENOMSG)
+                r = DHCP_ACK; // BOOTP messages don't have a DHCP message type option
```

Comment:
coding style: in the systemd tree we use `//` for "fixme" style comments and `/* */` for descripive comments like this one:

https://systemd.io/CODING_STYLE/#formatting